### PR TITLE
Clarify error message about leftclick on entity

### PR DESCRIPTION
**Pull request by oskarkk**
### Description
Change error message:

`A leftclick on an entity is an attack and thus not covered by the 'click' event, but the 'damage' event.`

to: 

`A leftclick on an entity is an attack and thus not covered by the 'click' event, but the 'damage' and 'vehicle damage' events.`

As I've suggested in issue #4429

---
**Target Minecraft Versions:** any
**Requirements:** none
**Related Issues:** #4429

### CODE REVIEW

**Comment by Pikachu920** [comment]
maybe we can do even better and only list the `vehicle damage` event if the entitydata is an instanceof `Vehicle`?

**Comment by AyhamAl-Ali** [comment]
> maybe we can do even better and only list the `vehicle damage` event if the entitydata is an instanceof `Vehicle`?

It would be nice to have this, makes the error message less in length since it's already long

**Comment by APickledWalrus** [comment]
> Ok, I'll try to do it.

Any update on this? Thank you :)

**Review by TheLimeGlass — APPROVED** [review]

File: src/main/java/ch/njol/skript/events/EvtClick.java [hunk 1]
```diff
@@ -31,43 +32,40 @@
...
-	
+
 	static {
 		Class<? extends PlayerEvent>[] eventTypes = CollectionUtils.array(
 			PlayerInteractEvent.class, PlayerInteractEntityEvent.class, PlayerInteractAtEntityEvent.class
 		);
-		
 		Skript.registerEvent("Click", EvtClick.class, eventTypes,
 				"[(" + RIGHT + "¦right|" + LEFT + "¦left)(| |-)][mouse(| |-)]click[ing] [on %-entitydata/itemtype%] [(with|using|holding) %-itemtype%]",
 				"[(" + RIGHT + "¦right|" + LEFT + "¦left)(| |-)][mouse(| |-)]click[ing] (with|using|holding) %itemtype% on %entitydata/itemtype%")
```
Comment (kiip1) [inline src/main/java/ch/njol/skript/events/EvtClick.java:71]:
```suggestion
				"[(" + RIGHT + ":right|" + LEFT + ":left)(| |-)][mouse(| |-)]click[ing] [on %-entitydata/itemtype%] [(with|using|holding) %-itemtype%]",
				"[(" + RIGHT + ":right|" + LEFT + ":left)(| |-)][mouse(| |-)]click[ing] (with|using|holding) %itemtype% on %entitydata/itemtype%")
```

File: src/main/java/ch/njol/skript/events/EvtClick.java [hunk 2]
```diff
@@ -31,43 +32,40 @@
...
 import ch.njol.util.Checker;
 import ch.njol.util.coll.CollectionUtils;
 
-@SuppressWarnings("unchecked")
 public class EvtClick extends SkriptEvent {
 
 	/**
 	 * Click types.
 	 */
 	private final static int RIGHT = 1, LEFT = 2, ANY = RIGHT | LEFT;
```
Comment (kiip1) [inline src/main/java/ch/njol/skript/events/EvtClick.java:53]:
```suggestion
	private enum ClickType {
		RIGHT,
		LEFT,
		ANY
	}
```

Comment (kiip1) [reply src/main/java/ch/njol/skript/events/EvtClick.java:53]: Then use a field, `ordinal()` or `name().toLowerCase()`

**Review by kiip1 — APPROVED** [review]

File: src/main/java/ch/njol/skript/events/EvtClick.java (hunk 2 above)
Comment (AyhamAl-Ali) [reply src/main/java/ch/njol/skript/events/EvtClick.java:53]: Would be nice! but it's out-of-scope for now, maybe in another PR \:)

**Review by AyhamAl-Ali — APPROVED** [review]

**Pull request merged**